src/containers/org/DataSetPermissionsContainer.js

```diff
@@ -58,18 +62,19 @@ const DataSetPermissionsContainer = ({
 |}) => {
 
   const dispatch = useDispatch();
+  const [keys, setKeys] = useState(List());
   const [paginationIndex, setPaginationIndex] = useState(0);
   const [paginationPage, setPaginationPage] = useState(0);
 
   const getDataSetPermissionsRS :?RequestState = useRequestState([PERMISSIONS, GET_DATA_SET_PERMISSIONS]);
+  const searchDataSetsRS :?RequestState = useRequestState([SEARCH, SEARCH_DATA_SETS]);
 
   const atlasDataSetIds :Set<UUID> = useSelector(selectOrganizationAtlasDataSetIds(organizationId));
   const atlasDataSetIdsHash :number = atlasDataSetIds.hashCode();
   const entitySetIds :Set<UUID> = useSelector(selectOrganizationEntitySetIds(organizationId));
   const entitySetIdsHash :number = entitySetIds.hashCode();
-  const keys :List<List<UUID>> = useMemo(() => (
-    Set().union(atlasDataSetIds).union(entitySetIds).map((id :UUID) => List([id]))
-  ), [atlasDataSetIdsHash, entitySetIdsHash]);
+  const searchHits :Set<UUID> = useSelector(selectDataSetSearchHits());
+  const searchHitsHash :number = searchHits.hashCode();
```

Comment:
just a thought, but it might be worth trying an equality function for the useSelector.

https://react-redux.js.org/api/hooks#equality-comparisons-and-updates